### PR TITLE
Add iOS Emoji Search sample app

### PR DESCRIPTION
**Pull request by efirestone**

### CODE REVIEW

File: samples/emoji-search/ios/app/EmojiSearchApp/RemoteImageLoader.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    private var cache: [URL: UIImage] = [:]
+
+    func loadImage(url: URL, completion: @escaping (URL, UIImage) -> Void) {
+        if let image = cache[url] {
+            completion(url, image)
+            return
+        }
+
+        // This is not a good image loader, but it's a good-enough image loader.
```
Comment (JakeWharton) [inline samples/emoji-search/ios/app/EmojiSearchApp/RemoteImageLoader.swift:18]: Love it

File: samples/emoji-search/ios/shared/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import org.jetbrains.kotlin.gradle.plugin.PLUGIN_CLASSPATH_CONFIGURATION_NAME
+
+plugins {
+  kotlin("multiplatform")
+  kotlin("native.cocoapods")
+}
+
+kotlin {
+  ios()
+  iosSimulatorArm64("ios")
```
Comment (JakeWharton) [inline samples/emoji-search/ios/shared/build.gradle.kts]: I will push to your/my/our branch and fix this up. I don't know why they didn't change the `ios()` shortcut to include the new simulator target.

**Review by JakeWharton — APPROVED** [review]

File: samples/emoji-search/ios/shared/build.gradle.kts (hunk 2 above)
Comment (JakeWharton) [reply samples/emoji-search/ios/shared/build.gradle.kts]: It looks like this builds, though. Bonkers!

I _think_ you're overriding the shared ios source set that targets actual devices and the x64 simulator with just the arm64 simulator which is why it works

Comment (JakeWharton) [reply samples/emoji-search/ios/shared/build.gradle.kts]: > I don't know why they didn't change the `ios()` shortcut to include the new simulator target.

https://youtrack.jetbrains.com/issue/KT-50615/

**Comment by JakeWharton** [comment]
I made it worse. Will fix in AM. I can't pod on my new computer so I just guessed.

**Comment by kpgalligan** [comment]
Need to add `import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget` to `samples/emoji-search/presenters/build.gradle.kts`. I can't push directly, so figured it was easier to describe in text than fork and PR :)

**Comment by JakeWharton** [comment]
Thanks. I pushed immediately before going to bed so figured it was something simple.

File: samples/emoji-search/ios/app/EmojiSearchApp/HTTPClient.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import shared
+import Foundation
+
+final class HTTPClient: Zipline_loaderZiplineHttpClient {
```
Comment (swankjesse) [inline samples/emoji-search/ios/app/EmojiSearchApp/HTTPClient.swift:4]: This is nice to have

File: samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    func httpCall(url: String, headers: [String : String], completionHandler: @escaping (String?, Error?) -> Void) {
+        var request = URLRequest(url: URL(string: url)!)
+        for (name, value) in headers {
+            request.addValue(value, forHTTPHeaderField: name)
+        }
+        let task = client.dataTask(with: request) { data, response, error in
+            // The KMM memory model doesn't do shared objects well, so Zipline expects the callback
+            // on the same thread that the download was initiated from. This happens to be the main
+            // thread, so we can bounce back to that thread for now.
+            // Switching to the new KMM memory model may remove the need for this.
```
Comment (swankjesse) [inline samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift:16]: Nice call out

File: samples/emoji-search/presenters/src/iosMain/kotlin/app/cash/zipline/samples/emojisearch/EmojiSearchZipline.kt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import app.cash.zipline.loader.ZiplineHttpClient
+import app.cash.zipline.loader.ZiplineLoader
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onEach
+import platform.Foundation.NSDate
+import platform.Foundation.timeIntervalSince1970
+
+class EmojiSearchZipline(
```
Comment (swankjesse) [inline samples/emoji-search/presenters/src/iosMain/kotlin/app/cash/zipline/samples/emojisearch/EmojiSearchZipline.kt:28]: I’ll take it as a follow-up task for me to potentially reuse more code here

File: samples/emoji-search/presenters/src/iosMain/kotlin/app/cash/zipline/samples/emojisearch/EmojiSearchZipline.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    value = EmojiSearchEvent.SearchTermEvent(searchTerm = "")
+  )
+  private var modelFlow: Flow<EmojiSearchViewModel>? = null
+
+  private val manifestUrl = "http://localhost:8080/manifest.zipline.json"
+
+  private val ziplineLoader = ZiplineLoader(
+    dispatcher = dispatcher,
+    httpClient = httpClient,
+    nowEpochMs = { NSDate().timeIntervalSince1970().toLong() * 1000 },
```
Comment (swankjesse) [inline samples/emoji-search/presenters/src/iosMain/kotlin/app/cash/zipline/samples/emojisearch/EmojiSearchZipline.kt:44]: For example, we can probably make this the default value for this!

**Review by swankjesse — APPROVED** [review]
# 🚀

File: samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift (hunk 4 above)
Comment (efirestone) [reply samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift:16]: I think this was copy/pasted from some older code. We should try again now that we're using the new memory model.

Comment (JakeWharton) [reply samples/emoji-search/ios/app/EmojiSearchApp/IosHostApi.swift:16]: Yeah I got it from our internal HTTP client

**Pull request merged**